### ISSUE TITLE
[Proposal] Windows App SDK projects should be buildable with `dotnet` CLI

### ISSUE THREAD

**Issue by kant2002** [microsoft/microsoft-ui-xaml#5327]
1. Create blank template application
2. Open folder for App1 in the x64 Dev tools
3. run `dotnet run`

```
...\nuget\packages\microsoft.projectreunion.foundation\0.8.0\build\Microsoft.ApplicationModel.Resources.PriGen.targets(386,5): error MSB4062: The "Microsoft.Build.AppxPackage.GetSdkFileFullPath" task could not be loaded from the assembly C:\Program Files\dotnet\sdk\6.0.100-preview.6.21315.23\\Microsoft\VisualStudio\v16.0\AppxPackage\Microsoft.Build.AppxPackage.dll. Could not load file or assembly 'C:\Program Files\dotnet\sdk\6.0.100-preview.6.21315.23\Microsoft\VisualStudio\v16.0\AppxPackage\Microsoft.Build.AppxPackage.dll'. The system cannot find the path specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [....\repos\App1\App1\App1\App1.csproj]
```

**Comment by StephenLPeters** [comment]
@alwu-msft do we support building with the dotnet command?

**Comment by alwu-msft** [comment]
No, not for Windows App SDK 1.0.

**Comment by kant2002** [comment]
Will you accept any contribution to that area? 
I personally want that, so I can add support for [NativeAOT](https://github.com/dotnet/runtimelab/issues/306)
If I can build using full blown MSBuild outside of VisualStudio that would be fine probably for me. 
If source code for build process open sourced I can hack around and contribute on improving build story.  So I would like to be pointed to approximate location where I can start looking. 
If this is somehow depends on the external projects (Nuget, MSBuild, whatever) I can go extra mile and start unblocking something in other location. But I need reference to that location and maybe a bit of direction. 

So probably my other question would be, will you positively look on this adventure and give me a bit of helping hand?

**Comment by alwu-msft** [comment]
To clarify my earlier statement, building with `msbuild` on the command line works just fine; it's the `dotnet` CLI specifically that we do not yet support. Unfortunately, many of the build tasks we depend on are not compatible with `dotnet` CLI as they target .NET Framework rather than .NET Standard, and as they are closed source (and I can't speak to when they might be open sourced, if ever) I don't think this is something the community can help us with. :(

**Comment by JensNordenbro** [comment]
@evelynwu-msft ; does things like PublishReadyToRun work ?

**Comment by evelynwu-msft** [comment]
@JensNordenbro Yes, `PublishReadyToRun` works.

**Comment by evelynwu-msft** [comment]
`dotnet *build*` works as of WinAppSdk 1.0.3. `dotnet *run*` does not, nor is there a definitive timeline on when it will.

@kant2002 I'll leave it up to you to determine if that satisfies your original feature request. :)

**Comment by kant2002** [comment]
Thank you! Now it’s time to check how things working and open new issue if needed.

**Comment by ianeuman** [comment]
I'm still experiencing this same error with dotnet build on WinAppSdk 1.0.4, do you want me to open a new issue?

```
C:\Users\ianeuman\.nuget\packages\microsoft.windowsappsdk\1.0.4\buildTransitive\MrtCore.PriGen.targets(360,5): error MSB4062: The "Microsoft.Build.AppxPackage.GetSdkFileFullPath" task could not be lo 
aded from the assembly C:\Program Files\dotnet\sdk\7.0.102\\Microsoft\VisualStudio\v17.0\AppxPackage\Microsoft.Build.AppxPackage.dll. Could not load file or assembly 'C:\Program Files\dotnet\sdk\7.0. 
102\Microsoft\VisualStudio\v17.0\AppxPackage\Microsoft.Build.AppxPackage.dll'. The system cannot find the path specified. Confirm that the <UsingTask> declaration is correct, that the assembly and al
l its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask.
```

**Comment by evelynwu-msft** [comment]
@ianeuman Yes, please open a new issue in the WinAppSDK repo: https://github.com/microsoft/WindowsAppSDK/
